torch/_dynamo/utils.py

```diff
@@ -799,6 +805,105 @@ def get_compilation_metrics() -> List[Union[CompilationMetrics, BwdCompilationMe
+        Logs the start of a single event.
+        :param str event_name Name of event to appear in trace
+        :param time_ns Timestamp in nanoseconds
+        :param metadata: Any extra metadata associated with this event
+        """
+        ChromiumEventLogger._log_timed_event(
+            event_name,
+            time_ns,
+            "B",
+            metadata,
+        )
+
+    @staticmethod
+    def log_event_end(
+        event_name: str,
+        time_ns: int,
+        metadata: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Logs the end of a single event. This function should only be
+        called after log_event_start with the same event_name.
+        :param event_name: Name of event to appear in trace
+        :param time_ns: Timestamp in nanoseconds
+        :param metadata: Any extra metadata associated with this event
+        """
+        ChromiumEventLogger._log_timed_event(
+            event_name,
+            time_ns,
+            "E",
+            metadata,
+        )
+
+    @staticmethod
+    def _log_timed_event(
+        event_name: str,
+        time_ns: int,
+        phase: str,
+        metadata: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Logs a timed event in chromium format. See log_event_start, log_event_end, etc.
+        """
+        event = {
+            "name": event_name,
+            "ts": time_ns / 1000,  # Chromium events are in ms
+            "args": metadata,
+            "ph": phase,
+            "pid": 0,  # pid should be specified on all logs, we don't personally care about the actual process id
+        }
+        torch._logging.trace_structured(
+            "chromium_event",
+            payload_fn=lambda: event,
+            suppress_context=False,
+            expect_trace_id=False,  # Not every chromium event will have a trace_id
+        )
+
+    @staticmethod
+    def log_instant_event(
+        event_name: str,
+        time_ns: int,
+        metadata: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Log an instant event with no associated duration.
+        :param str event_name: Name of event to appear in trace
+        :param int time_ns Timestamp in nanoseconds
+        :param Optional[Dict[str, Any]] metadata: Any extra metadata associated with this event
+        :param str cname optional color for the arrow in the trace
+        """
+        event = {
+            "name": event_name,
+            "ts": time_ns / 1000,
+            "args": metadata,
+            "ph": "i",
+            "pid": 0,  # pid should be specified on all logs, we don't personally care about the actual process id
+            "s": "p",  # We use "process" level instant events so they all appear on the same row in the trace.
+        }
+        torch._logging.trace_structured(
+            "chromium_event",
+            payload_fn=lambda: event,
+            suppress_context=False,
+            expect_trace_id=True,
+        )
```

Comment:
A schematic comment here. Because you're outputting into our proprietary structured trace format, there is no way to directly run event viewer on the log. Instead, you still need tlparse to postprocess the information for you. That means that it doesn't /really/ matter if you follow chrome trace event format here, because tlparse can always fix it as necessary. Especially your questions about pid/thread, that's something that tlparse could make decisions about, assuming you put enough metadata in these events to do it.

I think the real crux point here is that you want the trace format here to be Scuba compatible, because we're going to start shoving these in Scuba (most minimally, if you don't make a new table, into @oulgen's trace structured events table), and you want them to be in the right format to be scuba friendly since there's no postprocess step there.

Reply:
Your comment about protobuf suggests you might want to directly emit these traces at some point. Maybe that would be useful. But it's not what we have here...

Reply:
This is a good point: I think the structure I was gonna go for re: scuba was to implement a subclass of ChromiumEventLogger that, internally, will also log the same data to scuba. So the only interface that's important to be the same would be the arguments to `log_timed_event`, and then we could massage the inputs into whatever format we want in Scuba vs. chromium. 

A more universal format might be good anyway, though. The issue is, based on my experience with scuba, I think you kind of have to log each column separately, which means it's not ever really gonna be one exact output format anyway... I'll think on it